docs/docs/concepts.mdx

```diff
@@ -496,35 +501,56 @@ For specifics on how to use retrievers, see the [relevant how-to guides here](/d
 ### Tools
 <span data-heading-keywords="tool,tools"></span>
 
-Tools are interfaces that an agent, a chain, or a chat model / LLM can use to interact with the world.
+Tools are utilities designed to be called by a model: their inputs are designed to be generated by models, and their outputs are designed to be passed back to models.
+Tools are needed whenever you want a model to control parts of your code or call out to external APIs.
+
+A tool consists of:
+
+1. The name of the tool.
+2. A description of what the tool does.
+3. A JSON schema defining the inputs to the tool.
+4. A function (and, optionally, an async variant of the function).
+
+When a tool is bound to a model, the name, description and JSON schema are provided as context to the model.
+Given a list of tools and a set of instructions, a model can request to call one or more tools with specific inputs.
+Once the chosen tools are invoked, the results can be passed back to the model so that it can complete whatever task
+it's performing.
+
+#### Tool inputs
+
+A tool can take arbitrary arguments as input. At runtime, these arguments should be passed in as a dict to `tool.invoke({...})`.
```

Comment:
i would maybe add even more concrete examples of what this may look:
```
output = llm.bind_tools().invoke(...)
# check if tool call
tool_call = output.tool_calls[0]
tool.invoke(**tool_call['args'])
```